firebase/rules/tests/users.spec.ts

```diff
@@ -0,0 +1,89 @@
+import 'jest';
+import * as matchers from '../matchers';
+import * as targaryen from 'targaryen';
+
+const anonymousUser = targaryen.util.users.anonymous;
+const rules = require('../database.rules.json');
+
+describe('/users', () => {
+  let currentUser;
+  let systemUser;
+
+  beforeEach(() => {
+    jest.addMatchers(matchers);
+
+    targaryen.util.setFirebaseData({
+      users: {
+        '1': {
+          'common': {
+            id: '1',
+            displayName: 'Foo Bar',
+            email: 'foo.bar@example.com',
+            bio: '',
+            isAnonymous: false,
+            photoUrl: 'avatars.com/1'
+          }
+        },
+        '2': {
+          'common': {
+            id: '2',
+            displayName: 'Fizz Buzz',
+            email: 'fizz.buzz@example.com',
+            bio: '',
+            isAnonymous: false,
+            photoUrl: 'avatars.com/2'
+          }
+        }
+      }
+    });
+
+    currentUser = { uid: '1', email: 'foo@bar.com', provider: 'github' };
+    systemUser = { uid: 'SYSTEM_USER' };
+
+    targaryen.util.setFirebaseRules(rules);
+  });
+
+  it('isAnonymous is only true when the user is anonymous', () => {
+    expect(currentUser).cannotWrite('/users/1/common/isAnonymous', true);
```

Comment:
I think the counter proof is not necessary, because anonymous users are not stored in the `/users` node and only persisted once the user logs in with Github.